### ISSUE TITLE
DownloadResults with ZERO WIDTH NO-BREAK SPACE

### ISSUE THREAD

**Issue by stravag** [abbyysdk/ocrsdk.com#52]
Hi

We've been using the Android sample code to communicate with ABBY OCR. But we noticed that in the plain/text result of the `downloadResults` request at the beginning of the text there's always a ZERO WIDTH NO-BREAK SPACE (U+FEFF) character that we had to remove.

Why is that character in the response?

Our current "solution" is to remove the character in the `ResultsActivity`

`contents.toString().replaceFirst("\uFEFF", "")`

**Comment by Dmitry-Me** [comment]
I assume that you're describing the following scenario:
* your application invokes `getTaskStatus` and obtains an XML describing a completed task
* that XML contains a URL of a result file
* you download said result and it's a Unicode text file starting with `0xFEFF`

In this case the `FEFF` is a byte order mark (https://en.wikipedia.org/wiki/Byte_order_mark) indicating that it's UTF-16 big endian.

**Comment by stravag** [comment]
Yes thanks for the reply, that's exactly it! After further research I found that there is an old known bug in Java, it can't deal with BOMs automatically, making it necessary to remove it manually.

It would be good if you could change the sample code to incorporate that. https://github.com/abbyysdk/ocrsdk.com/blob/master/Android/src/abbyy/ocrsdk/android/ResultsActivity.java 

Apache Commons IO for instance provides a BOMInputStream

**Comment by Dmitry-Me** [comment]
Which `exportFormat` do you use when you invoke `/processImage`?

**Comment by stravag** [comment]
TXT Format: `language=English&exportFormat=txt`

**Comment by Dmitry-Me** [comment]
Where's the `downloadResults` method in the sample code?

**Comment by stravag** [comment]
https://github.com/abbyysdk/ocrsdk.com/blob/master/Android/src/com/abbyy/ocrsdk/Client.java#L153

**Comment by Dmitry-Me** [comment]
It's `donloadResult` (no `s`), that's why I asked. This method is for any type of results, including PDF, RTF and whatever else. Also even if it's a text result there's no guarantee that it will then be read and processed with Java code. So I'm sure it's a good idea to do anything with the output file in the generic Android sample.
